### PR TITLE
python3Packages.dulwich: 0.21.5 -> 0.21.6

### PR DESCRIPTION
**Pull request by schnusch**
Changelog: https://github.com/dulwich/dulwich/blob/dulwich-0.21.6/NEWS

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
That isn't really a mass rebuild, I think?

**Comment by schnusch** [comment]
I agree, a grep does not show too many occurences:

```sh
$ grep -Flr dulwich pkgs | sort | xargs grep -Fc dulwich
pkgs/applications/networking/errbot/default.nix:1
pkgs/applications/version-management/rabbitvcs/default.nix:1
pkgs/development/python-modules/breezy/default.nix:2
pkgs/development/python-modules/dulwich/default.nix:5
pkgs/development/python-modules/dvc-studio-client/default.nix:2
pkgs/development/python-modules/hg-git/default.nix:2
pkgs/development/python-modules/klaus/default.nix:2
pkgs/development/python-modules/openstackdocstheme/default.nix:2
pkgs/development/python-modules/scmrepo/default.nix:2
pkgs/development/python-modules/simplekv/default.nix:2
pkgs/development/tools/poetry2nix/poetry2nix/overrides/build-systems.json:1
pkgs/development/tools/poetry2nix/poetry2nix/pkgs/poetry/poetry.lock:64
pkgs/development/tools/poetry2nix/poetry2nix/pkgs/poetry/pyproject.toml:1
pkgs/development/tools/reno/default.nix:1
pkgs/servers/web-apps/netbox/generic.nix:1
pkgs/servers/xandikos/default.nix:1
pkgs/tools/package-management/poetry/unwrapped.nix:2
pkgs/top-level/python-packages.nix:1
```

**Comment by wegank** [comment]
@ofborg build xandikos

File: pkgs/development/python-modules/dulwich/default.nix [hunk 1]
```diff
@@ -17,15 +17,15 @@
 }:
 
 buildPythonPackage rec {
-  version = "0.21.5";
+  version = "0.21.6";
   pname = "dulwich";
   format = "setuptools";
 
   disabled = pythonOlder "3.6";
```
Comment (natsukium) [inline pkgs/development/python-modules/dulwich/default.nix:24]:
```suggestion
  disabled = pythonOlder "3.7";
```
https://github.com/jelmer/dulwich/blob/daf70786f929e3fd4995ad5d56bd93b437c20e5e/pyproject.toml#L26

**Review by natsukium — APPROVED** [review]

**Comment by wegank** [comment]
Superseded by #253612.

**Pull request closed without merging**